src/test/java/org/wmn4j/io/musicxml/MusicXmlWriterDomTest.java

```diff
@@ -4,21 +4,145 @@
 package org.wmn4j.io.musicxml;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.wmn4j.io.ParsingFailureException;
 import org.wmn4j.notation.TestHelper;
+import org.wmn4j.notation.builders.MeasureBuilder;
+import org.wmn4j.notation.builders.NoteBuilder;
+import org.wmn4j.notation.builders.PartBuilder;
+import org.wmn4j.notation.builders.ScoreBuilder;
+import org.wmn4j.notation.elements.Durations;
+import org.wmn4j.notation.elements.Note;
+import org.wmn4j.notation.elements.Pitch;
 import org.wmn4j.notation.elements.Score;
+import org.wmn4j.notation.iterators.PartWiseScoreIterator;
+import org.wmn4j.notation.iterators.ScoreIterator;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
 
 class MusicXmlWriterDomTest {
 
-	MusicXmlWriterDomTest() {
+	@TempDir
+	Path temporaryDirectory;
+
+	private static final String MUSICXML_FILE_PATH = "musicxml/";
+
+
+	private Score readMusicXmlTestFile(String testFileName, boolean validate) {
+		final MusicXmlReader reader = new MusicXmlReaderDom(validate);
+		Score score = null;
+		final Path path = Paths.get(TestHelper.TESTFILE_PATH + MUSICXML_FILE_PATH + testFileName);
+
+		try {
+			score = reader.readScore(path);
+		} catch (final IOException | ParsingFailureException e) {
+			fail("Parsing failed with exception " + e);
+		}
+
+		assertNotNull(score);
+		return score;
+	}
+
+	private Score writeScore(Score score) {
+		MusicXmlWriterDom writer = new MusicXmlWriterDom(score);
+		Path file = temporaryDirectory.resolve("file.xml");
+		writer.writeToFile(file.toString());
+
+		//For debugging
+		//printLines(file);
+
+		final MusicXmlReader reader = new MusicXmlReaderDom(true);
```

Comment:
This could use the MusicXmlReader.getReader method instead, so if the internal implementation of default MusicXML reader is changed, then this does not have to be changed.

Reply:
I Agree, I'll change this